libcalico-go/lib/clientv3/ippool_e2e_test.go

```diff
@@ -318,10 +339,57 @@ var _ = testutils.E2eDatastoreDescribe("IPPool tests", testutils.DatastoreAll, f
 			_, outError = c.IPPools().Get(ctx, name2, options.GetOptions{})
 			Expect(outError).To(HaveOccurred())
 			Expect(outError.Error()).To(ContainSubstring("resource does not exist: IPPool(" + name2 + ") with error:"))
+
+			By("Adding an IPPool with empty string for IPIPMode and VXLANMode and expecting it to be defaulted to 'Never'")
+
+			// Get the Calico backend client.
+			type accessor interface {
+				Backend() backendapi.Client
+			}
+			bc := c.(accessor).Backend()
+
+			// Add the IPPool through the backend so it skips setting default values
+			kvp := &model.KVPair{
+				Key: model.ResourceKey{
+					Name: name3,
+					Kind: apiv3.KindIPPool,
+				},
+				Value: &apiv3.IPPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              name3,
+						CreationTimestamp: metav1.Now(),
+						ResourceVersion:   "1",
```

Comment:
You shouldn't set `ResourceVersion` on Create() requests btw

Reply:
Strictly speaking, I don't think you need to specify CreationTimestamp or UID either - but you can if you want.

Reply:
I know I shouldn't/shouldn't need to, but `MatchResource()` on this line gives me trouble unless I set all of those:
https://github.com/projectcalico/calico/blob/c389512aba2ecf3548ae94486c66c319058ac45c/libcalico-go/lib/clientv3/ippool_e2e_test.go#L370

I could just use reflect to compare the pool's Spec, but I figured it would be good to keep the test consistent (and thus appease `MatchResource()` by including all of those). Let me know if you think that could be a problem. Thanks!

Reply:
Ah I see, yeah I guess that's OK. MatchResource is probably doing the wrong thing here but I think it's fine to leave it as you have it.